packages/flutter/lib/src/widgets/selectable_region.dart

```diff
@@ -1770,9 +1820,24 @@ abstract class MultiSelectableSelectionContainerDelegate extends SelectionContai
       }
     }
 
+    // Need to collect selection rects from selectables ranging from the
+    // currentSelectionStartIndex to the currentSelectionEndIndex.
+    final List<Rect> selectionRects = <Rect>[];
+    for (int index = currentSelectionStartIndex; index <= currentSelectionEndIndex; index++) {
+      final List<Rect>? currSelectableSelectionRects = selectables[index].value.selectionRects;
+      if (currSelectableSelectionRects != null) {
+        selectionRects.addAll(currSelectableSelectionRects.map((Rect selectionRect) {
+          final Matrix4 transform = selectables[index].getTransformTo(null);
```

Comment:
This should not be stored in global coordinates, but in local coordinates of `SelectionContainer`. 

Either everything is stored in global coordinates starting from the selectionGeomery of _SelectableFragment, or everything is stored in local coordinates of the `Selectable` including the SelectionContainer.

Since other properties in SelectionGeometry are in local coordinate, this should too. Use getTransformFrom(selectables[index]) instead.